yt_dlp/extractor/ted.py

```diff
@@ -26,120 +24,63 @@ class TEDIE(InfoExtractor):
-            'uploader_id': 'UCtelevision',
-            'upload_date': '20080522',
-        },
-        'params': {
-            'skip_download': True,
-        },
-    }, {
-        # no nativeDownloads
-        'url': 'https://www.ted.com/talks/tom_thum_the_orchestra_in_my_mouth',
-        'info_dict': {
-            'id': '1792',
-            'ext': 'mp4',
-            'title': 'The orchestra in my mouth',
-            'description': 'md5:5d1d78650e2f8dfcbb8ebee2951ac29a',
-            'uploader': 'Tom Thum',
+            'title': 'How to get serious about diversity and inclusion in the workplace',
+            'description': 'Imagine a workplace where people of all colors and races are able to climb every rung of the corporate ladder -- and where the lessons we learn about diversity at work actually transform the things we do, think and say outside the office. How do we get there? In this candid talk, inclusion advocate Janet Stovall shares a three-part action plan for creating workplaces where people feel safe and expected to be their unassimilated, authentic selves.',
             'view_count': int,
-            'comment_count': int,
-            'tags': list,
-        },
-        'params': {
-            'skip_download': True,
+            'tags': ['communication', 'community', 'work', 'humanity', 'race', 'social change', 'leadership', 'society', 'United States', 'equality'],
+            'uploader': 'Janet Stovall',
+            'duration': 664.0,
+            'upload_date': '20180822',
+            'release_date': '20180719',
+            'thumbnail': 'https://pi.tedcdn.com/r/talkstar-photos.s3.amazonaws.com/uploads/6a3549fa-2640-4ab4-892b-2491af834816/JanetStovall_2018S-embed.jpg',
         },
-    }, {
-        # with own formats and private Youtube external
-        'url': 'https://www.ted.com/talks/spencer_wells_a_family_tree_for_humanity',
-        'only_matching': True,
     }]
 
-    _NATIVE_FORMATS = {
-        'low': {'width': 320, 'height': 180},
-        'medium': {'width': 512, 'height': 288},
-        'high': {'width': 854, 'height': 480},
-    }
+    def _extract_info(self, webpage, video_name):
+        return self._parse_json(self._html_search_regex('<script[^>]+id="__NEXT_DATA__"[^>]*>(.+?)</script>', webpage, 'json'), video_name)
+
+    def _parse_playlist(self, playlist):
+
+        if isinstance(playlist, dict):
+            playlist_entries = []
 
-    def _extract_info(self, webpage):
-        info_json = self._search_regex(
-            r'(?s)q\(\s*"\w+.init"\s*,\s*({.+?})\)\s*</script>',
-            webpage, 'info json')
-        return json.loads(info_json)
+            for entry in try_get(playlist, lambda x: x['videos']['nodes'], list):
+                if entry.get('__typename') == 'Video':
+                    playlist_entries.append(self.url_result(entry.get('canonicalUrl'), self.ie_key()))
+
+            return playlist_entries
+        else:
+            return []
```

Comment:
```suggestion
    def _parse_playlist(self, playlist):
        if not isinstance(playlist, dict):
            return []
        return [
            self.url_result(entry['canonicalUrl'], self.ie_key())
            for entry in try_get(playlist, lambda x: x['videos']['nodes'], list) or []
            if entry.get('__typename') == 'Video']
```
Also, not sure if the isinstance check is actually needed